src/lib/y2users/linux/writer.rb

```diff
@@ -82,19 +82,19 @@ class Writer
       include Yast::Logger
       # Constructor
       #
-      # NOTE: right now we consider the system is empty, so we only receive one parameter.
-      #   But in the future we might need another configuration describing the initial state,
-      #   so we can compare and know what changes are actually needed.
-      #
-      # @param config [Y2User::Config] configuration containing the users
-      #   and groups that should exist in the system after writing
-      def initialize(config)
+      # @param config [Y2User::Config] see #config
+      # @param initial_config [Y2User::Config] see #initial_config
+      def initialize(config, initial_config)
```

Comment:
Maybe we can use system as default here so 

```suggestion
      def initialize(config, initial_config = Y2User::Config.system)
```

Reply:
I discussed it again with @ancorgs yesterday and the conclusion was to not have a default value (yet) since the initial_config might be different depending in the mode the code is being executed (installation, running system, auto upgrade, auto install, firstboot...).


Let's keep it as it is by now.